### PR TITLE
Add js unit tests

### PR DESCRIPTION
**Pull request by JeanPaiva**
This PR adds JS unit tests and the JS CI action.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,28 +22,21 @@
 		"dompurify": "^2.2.8"
 	},
 	"devDependencies": {
-		"@wordpress/api-fetch": "^5.0.0",
```
Comment (tomusborne) [inline package.json:25]: Any reason to remove all of these? They weren't necessary before but it seems like `wp-scripts` changed at some point and now requires that we install each one we need.

File: package.json [hunk 2]
```diff
@@ -22,28 +22,30 @@
...
+	  	"@wordpress/components": "^14.0.0",
+	  	"@wordpress/compose": "^4.0.0",
+	 	"@wordpress/data": "^5.0.0",
+	  	"@wordpress/element": "^3.0.0",
+	 	"@wordpress/hooks": "^3.0.0",
+	  	"@wordpress/i18n": "^4.0.0",
+	 	"@wordpress/rich-text": "^4.0.0",
+	  	"@wordpress/scripts": "^16.0.0",
+		"enzyme": "^3.11.0",
+		"grunt": "^1.2.1",
```
Comment (tomusborne) [inline package.json:37]: Did we need to downgrade this from 2.0.0? Same with other grunt packages?

Comment (JeanPaiva) [reply package.json:37]: Mine rebase messed up. I started this branch from master and switched later. I'll re-do my rebase and should fix all these issues.

**Review by tomusborne — APPROVED** [review]

File: .github/workflows/js-ci.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: GenerateBlocks JS CI
+
+on:
+  push:
+    branches:
+      - master
+      - 'release/**'
```
Comment (tomusborne) [inline .github/workflows/js-ci.yml:7]: GitHub is no longer letting me put `/` in the branch name - it gets converted to `-`.

**Pull request merged**